src/overlays/actors/ovl_En_Js/z_en_js.c

```diff
@@ -28,89 +39,1027 @@ const ActorInit En_Js_InitVars = {
+        Animation_Change(&this->skelAnime, &gMoonChildGettingUpAnim, -1.0f,
+                         Animation_GetLastFrame(&gMoonChildGettingUpAnim) - 1.0f, 0.0f, 2, -10.0f);
+        this->unk_2B8 &= ~0x10;
+        this->unk_2B8 |= 8;
+    }
+}
+
+s32 func_809695FC(EnJs* this, PlayState* play) {
+    if (ENJS_GET_EXIT_INDEX(&this->actor) == 0x3F) {
+        return false;
+    }
+    play->transitionTrigger = 0x14;
+    play->nextEntranceIndex = play->setupExitList[ENJS_GET_EXIT_INDEX(&this->actor)];
+    this->actionFunc = func_8096971C;
+    play->msgCtx.msgLength = 0;
+    gSaveContext.respawnFlag = -2;
+    return true;
+}
+
+void func_80969688(EnJs* this) {
+    if (this->cutsceneIndex != -1) {
+        if (ActorCutscene_GetCurrentIndex() == this->cutscenes[this->cutsceneIndex]) {
+            ActorCutscene_Stop(this->cutscenes[this->cutsceneIndex]);
+        }
+        this->cutsceneIndex = -1;
+    }
+}
+
+void func_809696EC(EnJs* this, s16 arg1) {
+    func_80969688(this);
+    this->cutsceneIndex = arg1;
+}
+
+void func_8096971C(EnJs* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+}
+
+void func_80969748(EnJs* this, PlayState* play) {
+    s32 item;
+    Player* player = GET_PLAYER(play);
+
+    SkelAnime_Update(&this->skelAnime);
+    Math_SmoothStepToS(&this->actor.world.rot.y, this->actor.yawTowardsPlayer, 6, 0x1838, 0x64);
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    if (Message_GetState(&play->msgCtx) == 0x10) {
+        item = func_80123810(play);
+        if (item != 0) {
+            this->actionFunc = func_80969898;
+        }
+        if (item > 0) {
+            func_801477B4(play);
+            if ((item >= PLAYER_AP_MASK_TRUTH) && (item <= PLAYER_AP_MASK_GIANT)) {
+                EnJs_TakeMask(item, ENJS_GET_TYPE(&this->actor));
+                Inventory_UnequipItem(item - 4);
+                if (!func_809692A8(ENJS_GET_TYPE(&this->actor))) {
+                    player->actor.textId = 0x2212;
+                } else {
+                    player->actor.textId = 0x2213;
+                }
+            } else if ((item >= PLAYER_AP_MASK_FIERCE_DEITY) && (item <= PLAYER_AP_MASK_DEKU)) {
+                player->actor.textId = 0x2211;
+            } else {
+                player->actor.textId = 0x2210;
+            }
+        }
+        if (item < 0) {
+            func_80151938(play, 0x2216);
+        }
+    }
+}
 
-extern ColliderCylinderInit D_8096ABA0;
+void func_80969898(EnJs* this, PlayState* play) {
+    s32 pad;
+    Player* player = GET_PLAYER(play);
 
-extern UNK_TYPE D_06010880;
-extern UNK_TYPE D_06016F58;
-extern UNK_TYPE D_06017E98;
+    SkelAnime_Update(&this->skelAnime);
+    Math_SmoothStepToS(&this->actor.world.rot.y, this->actor.yawTowardsPlayer, 6, 0x1838, 0x64);
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    switch (Message_GetState(&play->msgCtx)) {
+        case 4:
```

Comment:
```suggestion
        case TEXT_STATE_CHOICE:
```
New enum. Case 5 also should use the appropriate one.